lib/pacemaker/pcmk_ticket.c

```diff
@@ -83,3 +85,64 @@ pcmk_ticket_constraints(xmlNodePtr *xml, const char *ticket_id)
     pcmk__xml_output_finish(out, pcmk_rc2exitc(rc), xml);
     return rc;
 }
+
+int
+pcmk__ticket_get_attr(pcmk__output_t *out, pcmk_scheduler_t *scheduler,
+                      const char *ticket_id, const char *attr_name,
+                      const char *attr_default)
+{
+    int rc = pcmk_rc_ok;
+    const char *attr_value = NULL;
+    pcmk_ticket_t *ticket = NULL;
+
+    CRM_ASSERT(out != NULL && scheduler != NULL);
+
+    if (ticket_id == NULL || attr_name == NULL) {
+        return EINVAL;
+    }
+
+    ticket = g_hash_table_lookup(scheduler->tickets, ticket_id);
+
+    if (ticket != NULL) {
+        attr_value = g_hash_table_lookup(ticket->state, attr_name);
+    }
+
+    if (attr_value != NULL) {
+        out->message(out, "ticket-attribute", ticket_id, attr_name, attr_value);
+    } else if (attr_default != NULL) {
+        out->message(out, "ticket-attribute", ticket_id, attr_name, attr_default);
+    } else {
+        rc = ENXIO;
```

Comment:
Ideally we'd create new `pcmk_rc_*` values instead of ENXIO to get better log messages, but I'm fine with ignoring that for now